coremltools/converters/mil/mil/passes/defs/optimize_normalization.py

```diff
@@ -823,6 +836,190 @@ def _try_match_and_transform_pattern_4(self, reduce_op: Operation, block: Block)
             reduce_op, block, gamma_var, beta_var, epsilon_var, add_op, ops_to_remove
         )
 
+    def _try_match_and_transform_pattern_5(self, reduce_op, block) -> bool:
+        """
+        Detect BC1S ``LayerNorm`` pattern as in ml-ane-transformers.
+
+        Identify two patterns, the first:
+
+        ``y = (x - mean(x)) * rsqrt(variance(X) + eps)``
+
+        ``y = (x - mean(x)) * rsqrt(mean((x - mean(x))^2) + eps)``
+
+        .. code-block::
+
+            x --> reduce_mean --|
+            |                   |  |---|
+            |                   V  |   V
+            |----------------> sub --> mul --> reduce_mean --> add(epsilon) --> rsqrt
+                                |                                                   |
+                                |                                                   V
+                                |-------------------------------------------------> mul --> [...]
+
+        If the optional elementwise weight and bias are set, the second pattern is:
+
+        ``y = [(x - mean(x)) * rsqrt(mean((x - mean(x))^2) + eps) + beta] * gamma``
+
+        Note that this is different from the torch and MIL definitions of beta and gamma
+        so beta is be scaled by gamma and applied before it.
+
+        .. code-block::
+
+            x --> reduce_mean --|
+            |                   |  |---|
+            |                   V  |   V
+            |----------------> sub --> mul --> reduce_mean --> add(epsilon) --> rsqrt
+                                |                                                   |
+                                |                                                   V
+                                |-------------------------------------------------> mul
+                                                                                    |
+                                                                                    V
+                                                                                add(beta)
+                                                                                    |
+                                                                                    V
+                                                                                mul(gamma) --> [...]
+
+        These pattern corresponds to a specific ``layer_norm``:
+            - ``rank`` is 4.
+            - ``axes`` is ``[1]``
+            - ``gamma`` and ``beta`` are applied as in ml-ane-transformers, in the opposite order of torch.
+
+         """
+        ops_to_remove = []
+        root_var = reduce_op.x
+
+        if root_var.shape is None:
```

Comment:
I don't think there is such case where `Var.shape is None` ...
Did you hit any error without this line?

Reply:
I didn't hit any errors, I think I just copy-pasted this from the other patterns. Looks like they all do it.

Reply:
Yeah I think we should remove this logic ...
looks like a mistake we made before

Reply:
Removed it from this pattern.